frontend/taipy-gui/src/components/Taipy/TimeSelector.tsx

```diff
@@ -0,0 +1,111 @@
+ * the License. You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+import React, { useState, useMemo, useCallback} from "react";
+import Box from "@mui/material/Box";
+import Tooltip from "@mui/material/Tooltip";
+import { ErrorBoundary } from "react-error-boundary";
+import { TimePicker } from '@mui/x-date-pickers/TimePicker';
+import { isValid } from "date-fns";
+import { MobileTimePicker } from '@mui/x-date-pickers/MobileTimePicker';
+import { getSuffixedClassNames, TaipyActiveProps, TaipyChangeProps, getCssSize } from "./utils";
+import { createSendUpdateAction } from "../../context/taipyReducers";
+import { useClassNames, useDispatch, useDynamicProperty, useModule } from "../../utils/hooks";
+import ErrorFallback from "../../utils/ErrorBoundary";
+import Field from "./Field";
+
+
+interface TimeSelectorProps extends TaipyActiveProps, TaipyChangeProps {
+  asClock?: boolean;
+  format?: string;
+  defaultDate?: string;
+  defaultTime?: string;
+  defaultEditable?: boolean;
+  editable?: boolean;
+  label?: string;
+  time: string;
+  width?: string | number;
+}
+
+const boxSx = { display: "inline-block" };
+const TimeSelector = (props: TimeSelectorProps) => {
+  const { asClock = false, id, updateVarName, propagate = true } = props;
+  const [value, setValue] = useState <Date | null> (null);
+
+  const dispatch = useDispatch();
+  const module = useModule();
+  const className = useClassNames(props.libClassName, props.dynamicClassName, props.className);
+  const active = useDynamicProperty(props.active, props.defaultActive, true);
+  const editable = useDynamicProperty(props.editable, props.defaultEditable, true);
+  const hover = useDynamicProperty(props.hoverText, props.defaultHoverText, undefined);
+  const timeSx = useMemo(() => (props.width ? { maxWidth: getCssSize(props.width) } : undefined), [props.width]);
+
+  const handleChange = useCallback(
+    (v: Date | null) => {
+        setValue(v);
+        if (v !== null && isValid(v)) {
+            dispatch(
+              createSendUpdateAction(
+                  updateVarName,
+                  v,
+                  module,
+                  props.onChange,
+                  propagate
+              )
+            );
+        }
+    },
+    [updateVarName, dispatch, propagate, props.onChange, module]
+  );
+
+  return (
+    <ErrorBoundary FallbackComponent={ErrorFallback}>
+      <Tooltip title={hover || ""}>
+        <Box id={id} className={`${className}`} sx={boxSx}>
+          {editable ? (
+            asClock ? (
+              <MobileTimePicker
+                value={value}
```

Comment:
I don't see where you set the `value` from the `time` and `defaultTime` properties that come from the backend ?

Reply:
I follow the DateSelector's design pattern:

I passed the given ini value here:
const [value, setValue] = useState(() => getTime(props.defaultTime)); 

It was a bug (not able to pass the val frm BE to FE) indeed because wrong control default prop name, many thanks for your explicit clue